### PR TITLE
Add back variable logging for retention policies

### PR DESCRIPTION
**Pull request by brsm3129**
* **Tickets addressed:** MAXGNC-722
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

By creating new methods with similar functionality to the deprecated functions needed to log functions in parallelized simulations, we can better clarify where these should be used and properly phase out the old functions and variables.

## Verification

Added these functions to the MonteCarlo tests

## Documentation

bskPrinciples-11 still says the old functions are deprecated but we may want to add these functions to the MonteCarlo documentation

## Future work

Add tests that verify the contents of the logs instead of just verifying that they run.

### CODE REVIEW

File: src/utilities/SimulationBaseClass.py
Comment (patkenneally) [inline src/utilities/SimulationBaseClass.py]: :large_orange_diamond: We should change this to read "single process simulations"

File: src/utilities/SimulationBaseClass.py
Comment (patkenneally) [inline src/utilities/SimulationBaseClass.py]: :large_orange_diamond: Again "single process"

File: src/utilities/MonteCarlo/_UnitTests/test_scenarioBasicOrbitMC.py
Comment (patkenneally) [inline src/utilities/MonteCarlo/_UnitTests/test_scenarioBasicOrbitMC.py]: :large_orange_diamond: Should this test assert some invariant or check the values retrieved are as expected?

File: src/utilities/MonteCarlo/RetentionPolicy.py
Comment (patkenneally) [inline src/utilities/MonteCarlo/RetentionPolicy.py:102]: :large_orange_diamond: Were these unused?

**Review by patkenneally — APPROVED** [review]
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: src/utilities/MonteCarlo/RetentionPolicy.py
Comment (brsm3129) [reply src/utilities/MonteCarlo/RetentionPolicy.py:102]: Yes, totally unused.

File: src/utilities/MonteCarlo/_UnitTests/test_scenarioBasicOrbitMC.py
Comment (brsm3129) [reply src/utilities/MonteCarlo/_UnitTests/test_scenarioBasicOrbitMC.py]: Yes, currently they are just being checked to see if they run at all but I will look into seeing what we can assert. Likely making sure the shape of the output variable is greater than 2x0

Comment (brsm3129) [reply src/utilities/MonteCarlo/_UnitTests/test_scenarioBasicOrbitMC.py]: This was a great call because I totally realized I shouldnt have been adding logs to the sim directly, we now test in the same way as Messages and looks much better

**Pull request merged**
